cmd/monitorCodeGen/main.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/template"
+)
+
+var tmpl = `// Code generated by monitor-code-gen. DO NOT EDIT.
+
+package {{.goPackage}}
+
+import "github.com/signalfx/signalfx-agent/internal/monitors"
+
+{{with .groupMetricsMap}}
+const (
+{{- range . }}
+	{{(printf "group.%s" .Name) | formatVariable}} = "{{.Name}}"
+{{- end}}
+)
+{{end}}
+
+var groupSet = map[string]bool {
+{{- range .groupMetricsMap}}
+	{{(printf "group.%s" .Name) | formatVariable}}: true,
+{{- end}}
+}
+
+{{with .metrics}}
+const (
+{{- range .}}
+	{{.Name | formatVariable}} = "{{.Name}}"
+{{- with .Alias}}
+	{{. | formatVariable}} = "{{.}}"
+{{- end}}
+{{- end}}
+)
+{{end}}
+
+var metricSet = map[string]bool {
+{{- range .metrics}}
+	{{.Name | formatVariable}}: true,
+{{- end}}
+}
+
+var includedMetrics = map[string]bool {
+{{- range .metrics}}
+{{- if .Included}}
+	{{.Name | formatVariable}}: true,
+{{- with .Alias}}
+	{{. | formatVariable}}: true,
+{{- end}}
+{{- end}}
+{{- end}}
+}
+
+var groupMetricsMap = map[string][]string {
+{{- range $group, $metrics := .groupMetricsMap}}
+	{{(printf "group.%s" $group) | formatVariable}}: []string {
+		{{- range $metrics}}
+		{{. | formatVariable}},
+		{{- end}}
+	},
+{{- end}}
+}
+
+{{range .monitors}}
+var {{if gt (len $.monitors) 1 -}}
+{{- (printf "%s%s" .MonitorType "monitorMetadata") | formatVariable -}}
```

Comment:
"monitorMetadata" in this case should be capitalized right?

Reply:
There's not a need for it to be exported currently since it's passed as a reference via Register.

Reply:
But I mean if `.MonitorType` were, say, `cpu`, then wouldn't the variable name be `cpumonitorMetadata`?

Reply:
Yeah it would be wrong in that case. Fixed it by separating the monitor type from monitorMetadata with a period which `formatVariable` will then fixup.